### PR TITLE
Fixes PatternTerminal crafting issue

### PR DESCRIPTION
**Pull request by tyra314**
The pattern terminal doesn't craft at all. The problem is that `SlotCrafting::onTake` internally expects the inventory to be castable to `InventoryCraftResult`. However, it gets passed an `AppEngInternalInventory`. This replaces the `AppEngInternalInventory `with the vanilla class `InventoryCraftResult`.

This is a continuation and split of https://github.com/GuntherDW/Applied-Energistics-2/pull/10

This contains the **dangerous as well as a breaking** part.
TL;DR Some other mod may change the `InventoryCraftResult` class in a way that it doesn't respect the used fake items, potentially causing a dupe bug in AE2.

### CODE REVIEW

**Comment by yueh** [comment]
test this please.

File: src/main/java/appeng/container/implementations/ContainerPatternTerm.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import net.minecraft.inventory.IContainerListener;
 import net.minecraft.inventory.IInventory;
 import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.inventory.InventoryCraftResult;
```
Comment (orod-org) [inline src/main/java/appeng/container/implementations/ContainerPatternTerm.java]: MINOR Wrong order for 'net.minecraft.inventory.InventoryCraftResult' import. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.ImportOrderCheck)

File: src/main/java/appeng/container/implementations/ContainerPatternTerm.java [hunk 2]
```diff
@@ -415,6 +415,7 @@ public void craftOrGetItem( final PacketPatternSlot packetPatternSlot )
 
 			if( rr == r && Platform.itemComparisons().isSameItem( rr.getCraftingResult( real ), is ) )
 			{
+				this.cOut.setRecipeUsed(rr);
```
Comment (orod-org) [inline src/main/java/appeng/container/implementations/ContainerPatternTerm.java]: MINOR '(' is not followed by whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
MINOR ')' is not preceded with whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)

**Comment by orod-org** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by yueh** [comment]
So, I have looked at it and actually tested it (with a couple of different approaches).

Basically what gives me a headache is `getAndUpdateOutput` creating a local `InventoryCrafting`. As this is Algo's code, there are 2 cases. 
1. It is (or at least was) required for some important reason
2. It was a derp

Instead of changing the field, I simply created a local `InventoryCraftResult`, so `SlotCrafting#onTake()` would not fail, but it is still decoupled from any internal structures. Like

```
final InventoryCraftResult craftResult = new InventoryCraftResult();
craftResult.setRecipeUsed( rr );

final SlotCrafting sc = new SlotCrafting( p, real, craftResult, 0, 0, 0 );
sc.onTake( p, is );
```

And it appears to be working. Maybe there other issues with it, but for now I did not find any.

**Comment by tyra314** [comment]
Shall I update the PR so its in one place or do you want to make your own commit?

**Comment by yueh** [comment]
Feel free to update the PR. You probably spend way more time on it finding the actual bug.
